P2PEnabledConference.js

```diff
@@ -0,0 +1,899 @@
+     * for 'initiator' or 'responder'
+     * @return {FakeChatRoomLayer}
+     * @private
+     */
+    _createFakeRoom(isInitiator) {
+        return new FakeChatRoomLayer(this, isInitiator);
+    }
+
+    /**
+     * @inheritDoc
+     * @override
+     */
+    _doReplaceTrack (oldTrack, newTrack) {
+        const all = [super._doReplaceTrack(oldTrack, newTrack)];
+        if (this.peerToPeerSession) {
+            all.push(this.peerToPeerSession.replaceTrack(oldTrack, newTrack));
+        }
+        return Promise.all(all);
+    }
+
+    /**
+     * Intercepts events emitted by parent <tt>JitsiConference</tt>
+     * @private
+     */
+    _emitIntercept(eventType) {
+        const shouldBlock = this._shouldBlockEvent(eventType);
+        switch (eventType) {
+            // Log events which may be of interest for the P2P implementation
+            case JitsiConferenceEvents.CONNECTION_INTERRUPTED:
+            case JitsiConferenceEvents.CONNECTION_RESTORED:
+            case JitsiConferenceEvents.P2P_STATUS:
+                logger.debug(
+                    "_emitIntercept: block? " + shouldBlock, arguments);
+                break;
+        }
+        if (!shouldBlock)
+            this._originalEmit.apply(this.eventEmitter, arguments);
+    }
+
+    /**
+     * @inheritDoc
+     * @override
+     * @private
+     */
+    _ensureRemoteStatsRunning () {
+        if (this.p2pEstablished) {
+            // Use remote starts from the P2P connection
+            this.statistics.startRemoteStats(
+                this.peerToPeerSession.peerconnection);
+        } else {
+            // This will start remote stats for the JVB connection
+            super._ensureRemoteStatsRunning();
+        }
+    }
+
+    /**
+     * Called when {@link JitsiConferenceEvents.CONNECTION_ESTABLISHED} event is
+     * triggered for the P2P session. Switches the conference to use the P2P
+     * connection.
+     * @param {JingleSessionPC} jingleSession the session instance. It should be
+     * always the P2P one, but still worth to verify for bug detection
+     * @private
+     */
+    _onP2PConnectionEstablished (jingleSession) {
+        if (this.peerToPeerSession !== jingleSession) {
+            logger.error("CONNECTION_ESTABLISHED - not P2P session ?!");
+            return;
+        }
+        // Update P2P status and emit events
+        this._setNewP2PStatus(true);
+
+        // Remove remote tracks
+        this._removeRemoteJVBTracks();
+        // Add remote tracks
+        this._addP2PRemoteTracks();
+        // Remove local tracks from JVB PC
+        // But only if it has started
+        if (this.jingleSession) {
+            this._removeLocalTracksFromJVB();
+        }
+
+        // Start remote stats
+        logger.info("Starting remote stats with p2p connection");
+        this._ensureRemoteStatsRunning();
+    }
+
+    /**
+     * Detaches local tracks from the JVB connection.
+     * @private
+     */
+    _removeLocalTracksFromJVB() {
```

Comment:
i think it'd be a bit clearer if the method name here matched the description more closely.  e.g. `_removeLocalTracksFromJvbPeerConnection`